lib/RuleBase.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+/** 
+ * Helper class that allows client code to group evaluator rules in an array of 3-tuples,
+ * each 3-tuple associates the evaluator (built out of a rule) to its output and evaluation result.
+ * (For more information on to use it, please refer to example_1_RuleBase.js).
+*/
+class RuleBase {
+
+  /** 
+   * just creates an empty array to hold the 3-tuples 
+   * */
+  constructor() {
+    this.rules = [];
+  }
+
+  /** 
+   * adds a 3-tuples of { evaluator, evaluation, result } and the evaluation is 
+   * initialized to "false".
+   */
+  addRule(evaluator, result) {
+    this.rules.push( { evaluator: evaluator, evaluation: false, result: result} );
```

Comment:
will a max value for rules be useful here? just to make sure no one is creating something uncomputable

Reply:
I agree